sites/website/src/docs/design-systems/creating-a-component-library.md

```diff
@@ -0,0 +1,145 @@
+
+### Define Templates and Styles 
+
+With our `Counter` component class written, we now need to define its templates and styles. There are two ways to accomplish this. Just like `FASTElement`, a component that extends `FoundationElement` can be used with the [`html`](/docs/fast-element/declaring-templates) and [`css`](/docs/fast-element/leveraging-css) tagged template literals. So, we could define our template and styles in the same fashion as any other component, like this:
+
+```ts
+import { html, css  } from "@microsoft/fast-element";
+
+const counterStyles = css`/* ... */`;
+const counterTemplate = html`<!-- ... -->`;
+```
+
+However, a `FoundationElement` also supports a lazily-defined template and style type. This lazy option is a function invoked with [ElementDefinitionContext](/docs/api/fast-foundation.elementdefinitioncontext) and [FoundationElementDefinition](/docs/api/fast-foundation.foundationelementdefinition) options, providing opportunity to adjust the template or styles based on how the element or other elements are defined. Lazy template and style definitions provide incredible flexibility, and are what can allow application authors to re-name component tag names, override default slotted content, and communicate any component-specific configuration options. Here's how we would convert the previous standard templates and styles into lazy templates and styles:
+
+```ts
+import { html, css  } from "@microsoft/fast-element";
+import { ElementDefinitionContext, FoundationElementDefinition } from "@microsoft/fast-foundation";
+
+const counterStyles = (
+  context: ElementDefinitionContext,
+  definition: FoundationElementDefinition
+) => css`/* ... */`;
+
+const counterTemplate = (
+  context: ElementDefinitionContext,
+  definition: FoundationElementDefinition
+) => html`<!-- ... -->`;
+```
+
+Let’s take a closer look at what the two arguments to these lazily evaluated functions give us.
+
+#### What is the `ElementDefinitionContext`?
+
+The first argument is the `ElementDefinitionContext`. This context object provides information that is available *during* the attempt to register the element. For example, it will tell you the HTML element tag `name` that the component will be defined as. You can also see what the default `shadowRootMode` of the `DesignSystem` is. Most of the other APIs are used internally by `FoundationElement.compose` (see below) to define the element, but one particular API of note is the `tagFor()` method of the context. This API is used to inspect the tag-name of other components and allows usage of other library components in a template or stylesheet without knowing ahead of time what the tag-name for the element will be. Here's an example of how the reusable `Counter` component would leverage this to ensure that the `Button`'s tag name is correct, even if the app developer decides to rename `Button` in some way.
```

Comment:
would be great to also document how `tagFor` is used in the CSS context, like in https://github.com/microsoft/fast/pull/4963/files

Reply:
good call, updating.